### PR TITLE
[WIP] Initial Roadmap

### PR DESCRIPTION
**Pull request by mikeal**
Initial roadmap, adapted from early OKR's and vision statements.

### CODE REVIEW

**Comment by warpfork** [comment]
Do we want to put something in here about the Data Model spec being landed, specifically?

**Comment by mikeal** [comment]
Which one? The basic one is part of the spec refactor that is already
there. The future type system one is not there cause I dont know what
@stebalien's timeline is for that yet.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,32 @@
-# IPLD Roadmap
-
 IPLD Project Roadmap
 
-## Q4 2018
+# IPLD Vision
+
+Create the internet of data-structures.
```
Comment (vmx) [inline README.md]: Just want to double check if "internet" is intentionally lowercase or not.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,32 @@
-# IPLD Roadmap
-
 IPLD Project Roadmap
```
Comment (vmx) [inline README.md:1]: Should we remove this line completely, it looks strange to me having some text before the headline.

**Review by vmx — COMMENTED** [review]
Looks good, thanks for writing this up. I've just two minor comments.

**Comment by mikeal** [comment]
@vmx what do you want me to write in for GraphSync?

**Comment by vmx** [comment]
@mikeal I need to put more thought on this. I haven't had a closer look at Juan's proposal yet, I just watched the talk. At the moment I lean towards just putting our proposal (once it's out) on the roadmap. I don't want to commit to Juan's proposal.

File: README.md (hunk 2 above)
Comment (mikeal) [reply README.md:1]: The way GitHub shows the README text on the main page ends up displaying this rather nicely, it just doesn't look great in the raw markdown version :(

**Comment by mikeal** [comment]
@vmx I'm starting to think that we should pull back a little bit and re-scope all of this under the umbrella of "IPLD Graph Replication."

Graph replication should be in the vision statement and then we can break out specific projects for graph replication targeted at specific use cases.

Calling all of these competing proposals "GraphSync" without a clear problem statement anywhere is highly problematic. If it turns out that two of these proposals are great approaches but for different use cases we may actually want to have projects to implement both, but without an umbrella like "replication" we can't find the commonalities between them.

**Comment by vmx** [comment]
Not sure where to put it. I'm also involved in the JS IPFS Core roadmapping. There I've put in an item about adding search to the npm on IPFS project. This is basically https://github.com/ipld/ipld/issues/43. Perhaps we should link it in our roadmap once the JS IPFS Core one is finialised.

**Comment by mikeal** [comment]
I think we should merge this and begin iterating on it with additional pull requests for specific things like GraphSync.

**Review by warpfork — APPROVED** [review]

**Review by vmx — APPROVED** [review]

**Pull request merged**